lib/virtfs/block_io.rb

```diff
@@ -11,6 +11,8 @@ module VirtFS
   class BlockIO
     MIN_BLOCKS_TO_CACHE = 64
 
+    attr_reader :seek_pos
```

Comment:
@movitto Ruby's IO objects are thread safe within each operation (read, write, seek) but they're not thread safe if you're going to depend on any _state_ within an object shared between threads. That's why you shouldn't rely on the current position - or even use `SEEK_CUR` - outside the scope of a mutex mechanism. Between any IO operations performed by the FS, a different thread could perform different operations on the same underlying IO stream. Instead, if the FS were to maintain the state it needs in local variables, then only the `seek`, `read` operations need to be performed under mutex. So, relying on the IO object's current seek position not changing beyond the scope of a single method call, would make it very difficult to make the FS thread safe in the future.